src/main/java/in/projecteka/datanotificationsubscription/subscription/SubscriptionRequestRepository.java

```diff
@@ -50,17 +52,17 @@
             "(request_id, patient_id, status, details, requester_type) VALUES ($1, $2, $3, $4, $5)";
 
     private static final String INSERT_SOURCES_REQUEST_QUERY = "INSERT INTO subscription_source " +
-            "(subscription_id, period_from, period_to, category_link, category_data, hip_id, hi_types, status) VALUES ($1, $2, $3, $4, $5, $6, $7, $8)";
+            "(subscription_id, period_from, period_to, category_link, category_data, hip_id, hi_types, status, excluded) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)";
 
     private static final String GET_SUBSCRIPTION_REQUEST_QUERY = "SELECT details, request_id, status, date_created, date_modified, requester_type FROM "
             + "hiu_subscription WHERE patient_id=$1 and (status=$4 OR $4 IS NULL) " +
             "ORDER BY date_modified DESC" +
             " LIMIT $2 OFFSET $3";
 
     private static final String GET_ACTIVE_LINK_SUBSCRIPTION_QUERY = "SELECT hs.request_id, hs.patient_id, hs.subscription_id, " +
-            "hs.details -> 'hiu' -> 'id' AS hiu_id FROM hiu_subscription hs INNER JOIN subscription_source ss ON hs.subscription_id = ss.subscription_id WHERE " +
-            "hs.patient_id=$1 AND hs.status=$2 AND ss.hip_id=$3 AND ss.status=$4 AND ss.category_link=$5" +
-            " AND ss.period_from<=$6 AND ss.period_to>= $7";
+            "hs.details -> 'hiu' -> 'id' AS hiu_id, ss.hip_id, ss.excluded FROM hiu_subscription hs INNER JOIN subscription_source ss " +
+            "ON hs.subscription_id = ss.subscription_id WHERE hs.patient_id=$1 AND hs.status=$2 AND (ss.hip_id=$3 OR ss.hip_id='" + INCLUDE_ALL_HIPS_CODE + "')" +
```

Comment:
we can't leave ss.hip_id as blank?

Reply:
You mean instead of `ALL` use black in the database for this case? We can do that too. I will change and push in sometime.